generator/mavgen_java.py

```diff
@@ -93,9 +101,29 @@ def generate_CRC(directory, xml):
     *
     * @param msgid
     *            The message id number
+    * @return crc_extra
+    *            The crc_extra value for the given message
     */
-    public void finish_checksum(int msgid) {
-        update_checksum(MAVLINK_MESSAGE_CRCS[msgid]);
+    public int finish_checksum(int msgid) {
+        int crc = 0;
+        if (msgid == 0) {
```

Comment:
This should go away, heartbeat isn't actually a high frequency message on almost any deployed application, and is frankly not a parser concern.